src/subdomains/staking/application/services/liquidity-management.service.ts

```diff
@@ -104,8 +115,22 @@ export class LiquidityManagementService {
     return this.handleMasternodesWithState(masternodes.splice(0, count), MasternodeState.ENABLED);
   }
 
-  private async handleMasternodesWithState(masternodes: Masternode[], state: MasternodeState): Promise<void | void[]> {
-    const filteredMasternodes = masternodes.filter((mn) => mn.state === state);
+  private async handleMasternodesWithState(
+    masternodes: Masternode[],
+    state: MasternodeState,
+    blockchainState?: BlockchainMasternodeState,
+  ): Promise<void | void[]> {
+    let filteredMasternodes = masternodes.filter((mn) => mn.state === state);
+
+    if (blockchainState) {
+      filteredMasternodes = filteredMasternodes.filter(async (mn) => {
```

Comment:
I think filter does not work with an async method.

Reply:
There's already a similar logic in MasternodeService.getOrderedByTms. There we use the same method to get the current target multiplier of the MN. Perhaps we should create a method `filterByBlockchainState` or something there?